### PR TITLE
feature/117_whitespace_jinja_readme

### PR DESCRIPTION
**Pull request by damian-romero**
See #117

### CODE REVIEW

**Comment by damian-romero** [comment]
Sorry, @astrochun. I forgot to add you as a co-author on the commits...

File: ldcoolp/curation/inspection/readme/__init__.py [hunk 1]
```diff
@@ -253,10 +258,12 @@ def retrieve_article_metadata(self):
...
-        readme_dict['description'] = html2text(description)
-        # Strip extra white space from html2text
-        if readme_dict['description'][-2:] == "\n\n":
-            readme_dict['description'] = readme_dict['description'][:-2]
+        readme_dict['description'] = parser.handle(description)
+
+        # Clean whitespace around readme_dict keys
+        for k, v in readme_dict.items():
+            if isinstance(v, str):
+                readme_dict[k] = readme_dict[k].strip()
```
Comment (astrochun) [inline ldcoolp/curation/inspection/readme/__init__.py:267]: I like this change, but unclear why this is done twice (here) and below.

Comment (astrochun) [reply ldcoolp/curation/inspection/readme/__init__.py:267]: Ah, my bad. Just realized it was for two functions

**Review by astrochun — COMMENTED** [review]
This looks good. There may be more additional edge cases to consider. I wish we had a wider set of tests with various inputs for README to check.

**Comment by astrochun** [comment]
Per our discussion, @damian-romero and I decided to remove the html parsing and have that as part of a feature down the road.

File: ldcoolp/curation/inspection/readme/__init__.py (hunk 1 above)
Comment (damian-romero) [reply ldcoolp/curation/inspection/readme/__init__.py:267]: - Yes, I didn't think it was worth creating a method just for these two cases.
- I agree we need to have a specific set of deposits for edge cases.

File: ldcoolp/curation/inspection/readme/__init__.py [hunk 2]
```diff
@@ -18,6 +18,11 @@
 
 from ...api.qualtrics import Qualtrics
 
+# Start html parser for retrieve_article_metadata() method
+parser = html2text.HTML2Text()
+parser.body_width = 0  #  No line wrapping
```
Comment (damian-romero) [inline ldcoolp/curation/inspection/readme/__init__.py:23]:
```suggestion
```

**Review by astrochun — COMMENTED** [review]
With the HTML parsing, this is what I get for a test deposit that has rich metadata and whitespaces:

[README.txt](https://github.com/ualibraries/LD_Cool_P/files/5705929/README.txt)

Here is how it was before this feature (Note the differences in the Summary section):
[README.txt](https://github.com/ualibraries/LD_Cool_P/files/5705931/README.txt)

This is an edge case, but worth noting.

**Review by astrochun — CHANGES_REQUESTED** [review]
Changes needed to remove the HTML parsing

**Review by astrochun — APPROVED** [review]

**Review by astrochun — APPROVED** [review]
Changes look good.
